fdbserver/TLogServer.actor.cpp

```diff
@@ -1816,28 +1815,12 @@ Future<Void> tLogPeekMessages(PromiseType replyPromise,
 
 	// if tLog locked for recovery, return an empty message at the cluster recovery version
 	// if requested version is greater than any received.
-	state Optional<Version> clusterRecoveryVersion = Optional<Version>();
-	if (SERVER_KNOBS->ENABLE_VERSION_VECTOR_TLOG_UNICAST && logData->stopped()) {
-		DebugLogTraceEvent("TLogPeekMessagesWaitForCrv", self->dbgid)
-		    .detail("Tag", reqTag.toString())
-		    .detail("LogId", logData->logId)
-		    .detail("ReqReturnIfBlocked", reqReturnIfBlocked ? 1 : 0)
-		    .detail("ReqBegin", reqBegin)
-		    .detail("Version", logData->version.get())
-		    .detail("RecoveredAt", logData->recoveredAt);
-
-		if (!logData->clusterRecoveryVersion.get().present()) {
-			wait(timeoutError(logData->clusterRecoveryVersion.onChange(), 100.0));
-		}
-		// Obtain the first unreadable version number of the recovered log.
-		clusterRecoveryVersion = logData->clusterRecoveryVersion.get().get();
-		DebugLogTraceEvent("TLogPeekMessagesClusterRecoveryVersion", self->dbgid)
-		    .detail("LogId", logData->logId)
-		    .detail("LogEpoch", logData->epoch())
-		    .detail("CRV", clusterRecoveryVersion.get());
+	state bool clusterRecovery = false;
+	if (SERVER_KNOBS->ENABLE_VERSION_VECTOR_TLOG_UNICAST && logData->version.get() < reqBegin && logData->stopped()) {
```

Comment:
It appears to me you are checking for "logData->version.get() < reqBegin" everywhere "clusterRecovery" is checked, so maybe you can take that version check out of here?